src/main/java/no/difi/sdp/client2/internal/AddClientVersionInterceptor.java

```diff
@@ -1,35 +1,50 @@
 package no.difi.sdp.client2.internal;
 
+import no.difi.sdp.client2.domain.exceptions.SendIOException;
 import org.apache.http.Header;
 import org.apache.http.HttpException;
 import org.apache.http.HttpRequest;
 import org.apache.http.HttpRequestInterceptor;
 import org.apache.http.protocol.HttpContext;
 
 import java.io.IOException;
+import java.io.InputStream;
+import java.text.MessageFormat;
+import java.util.Properties;
 
 public class AddClientVersionInterceptor implements HttpRequestInterceptor {
 
     private final String clientVersion;
     private final String javaVersion;
 
     public AddClientVersionInterceptor() {
-        String implementationVersion = getClass().getPackage().getImplementationVersion();
+        this.clientVersion = getProjectVersion();
         String javaVersion = System.getProperty("java.version");
-        this.clientVersion = implementationVersion != null ? implementationVersion : "UNKNOWN";
         this.javaVersion = javaVersion != null ? javaVersion : "UNKNOWN";
     }
 
     @Override
     public void process(HttpRequest request, HttpContext context) throws HttpException, IOException {
         Header[] headers = request.getHeaders("User-Agent");
-        String clientUserAgent = "Java/" + javaVersion + " DifiSdp/" + clientVersion;
+        String clientUserAgent = MessageFormat.format("difi-sikker-digital-post-klient-java/{0} (Java/{1})", clientVersion, javaVersion);
+
         if (headers.length == 0) {
             request.addHeader("User-Agent", clientUserAgent);
-        }
-        else {
+        } else {
             request.addHeader("User-Agent", headers[0].getValue() + " " + clientUserAgent);
         }
     }
 
+    private String getProjectVersion() {
+        InputStream resourceAsStream = this.getClass().getResourceAsStream("/project.properties");
```

Comment:
Denne ressursen trenger ikke leses inn hver gang man gjør en request. Kan leses inn én gang, f.eks. i en static initializer for klassen.

Streamen må også lukkes.

Pro-tip: det bør sitte i ryggmargen og alltid "acquire" en `{Input|Output}Stream` på denne måten:
```java
try (InputStream in = gimmeZumStream()) {

}
```
(I tilfeller hvor man får en stream inn som en parameter til en metode skal man generelt _ikke_ gjøre noe ressurshåndtering (a.k.a. lukke streamen) inni metoden, da streamen er blitt "acquired" utenfor, og det er "noen andre" som har ansvaret for ressurshåndteringen.)